configure.ac

```diff
@@ -571,6 +571,7 @@ WRAPPER_COMPILER_EXTRA_LDFLAGS=`echo "$WRAPPER_COMPILER_EXTRA_LDFLAGS" | sed -e
 AC_MSG_RESULT([$WRAPPER_COMPILER_EXTRA_LDFLAGS])
 AC_SUBST(WRAPPER_COMPILER_EXTRA_LDFLAGS)
 AC_SUBST(WRAPPER_COMPILER_EXTRA_LIBS)
+AC_SUBST(PTHREAD_CFLAGS)
```

Comment:
Perhaps the confusion is due to the fact that the PTHREAD_* variables are not "preset output variables" like CFLAGS, LDFLAGS, and friends, so they are not defined in any Makefile.in by default. With AC_SUBST(PTHREAD_*), the @PTHREAD_*@ var is placed in Makefile.in and set at configure time.  On Ubuntu 16.04, opal_config_pthreads.m4 _only_ sets PTHREAD_CFLAGS for some reason, and the mt_a2a unit test needs -pthread at _link_ time to build.  So, another option is to include PTHREAD_CFLAGS in LDFLAGS in configure.ac.  Yet another option is to figure out why PTHREAD_LIBS and/or PTHREAD_LDFLAGS aren't set correctly.  I hope that makes sense...